File: PostsSPA/src/components/spa/post/models/paging-params.ts
import { environment } from 'src/environments/environment';

export class PagingParams {
  PageSize: number;
  PageNumber: number;
  IsOrderByLatest: boolean;

  constructor(params: PagingParams = {} as PagingParams) {
    const {
      PageSize = environment.defaultPageSize,
      PageNumber = 1,
      IsOrderByLatest = environment.defaultOrderByLatest
    } = params;

    this.PageSize = PageSize;
    this.PageNumber = PageNumber;
    this.IsOrderByLatest = IsOrderByLatest;
  }
}

File: PostsAPI/Models/PagingParams.cs
namespace PostsAPI.Models
{
    public class PagingParams
    {  
        const int MaxPageSize = 20; 
        private int pageSize  = 10; 
  
        public int PageNumber { get; set; } = 1;  

        public bool IsOrderByLatest { get; set; } = true;
  
        public int PageSize  
        {  
            get 
            { 
                return pageSize; 
            }  
            set  
            {  
                pageSize = (value > MaxPageSize) ? MaxPageSize : value;  
            }  
        }  
    }  

}